charms/kfp-schedwf/src/charm.py

```diff
@@ -58,12 +65,30 @@ def __init__(self, *args):
                 krh_labels=create_charm_default_labels(
                     self.app.name, self.model.name, scope="auth-and-crds"
                 ),
-                context_callable=lambda: {"app_name": self.app.name, "namespace": self._namespace},
+                context_callable=lambda: {
+                    "app_name": self.app.name,
+                    "namespace": self.model.name,
+                    "sa_name": SA_NAME,
+                },
                 lightkube_client=lightkube.Client(),
             ),
             depends_on=[self.leadership_gate],
         )
 
+        self.sa_token = self.charm_reconciler.add(
```

Comment:
suggestion(non-blocking): add a comment explaining what this component maps to in upstream